connect_ext_ppr/service.py

```diff
@@ -90,3 +97,136 @@ def update_product(data, config, logger):
             product.version = data['version']
             db.add(product)
             db.commit()
+
+
+def get_ppr_new_version(db, deployment):
+    version = (
+        db.query(PPRVersion.version)
+        .filter_by(deployment=deployment.id)
+        .order_by(desc(PPRVersion.version))
+        .with_for_update()
+        .limit(1)
+        .scalar()
+    ) or 0
+    new_version = version + 1
+    return new_version
+
+
+def create_ppr(ppr, context, deployment, db, client, logger):
```

Comment:
This function is very big and has a lot of logic. It might be more convenient to refactor it and to split to several separate functions.

Reply:
I agree here, at least there are two branches that can be done as separate functions --> uploading and generation